src/mega_lehmer.rs

```diff
@@ -1,12 +1,16 @@
-use crate::felt::FeltTrait;
+use crate::{felt::FeltTrait, seed_generators::time_seed};
 
 pub struct MegaLehmer<F: FeltTrait> {
     multiplier: F,
     last_gen: F,
 }
 
 impl<F: FeltTrait> MegaLehmer<F> {
-    pub fn new(seed: F, multiplier: F) -> MegaLehmer<F> {
+    pub fn new(seed: Option<F>, multiplier: F) -> MegaLehmer<F> {
+        let seed = match seed {
+            Some(seed) => seed,
+            None => time_seed(),
+        };
```

Comment:
Same as in my comment (what if we have more gens?)

Reply:
We first need a `SeedGeneratorTrait` for now, this is the only thing we have. We will probably use a generator as a param